android/src/main/AndroidManifest.xml

```diff
@@ -1,5 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="io.customer.reactnative.sdk">
+    
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
```

Comment:
hmm, I believe this is only available in Android SDK > 32. So it might be problematic to add it for the users who are below 32. 

 I came around this [document](https://developer.android.com/guide/topics/manifest/uses-sdk-element) that lets you restrict the SDK version but the warning below it is scary. 
 
 Can we probably ask customers to add it? because if its added in the `app` it won't have any issues even below 33.

Reply:
That's one of the call we have make.

**If we add this to the SDK:**

- Customers using compileSdkVersion 33 or greater -> Should not see any error
- Customers using compileSdkVersion less than 33 -> Will have to remove `POST_NOTIFICATIONS` permission using `tools:node`

**If we don't add this to the SDK:**

- Customers using `compileSdkVersion` 33 or greater -> Will have to add `POST_NOTIFICATIONS` to their `AndroidManifest`
- Customers using `compileSdkVersion` less than 33 -> Should not see any error

In most cases, I would generally vote for going with newer APIs (option 1 here). But here since we are still using `compileSdkVersion` `30` in both Android and React Native SDKs, I think asking the customers to add this permission is fine until we don't update our `compileSdkVersion` to `33`

Reply:
**Updates on adding permission to `AndroidManifest`:**

Here are the results from my testing on AmiApp locally:

- `targetSdkVersion = 31`, `compileSdkVersion = 31`, Device API Level 30 -> Permission granted by default
- `targetSdkVersion = 31`, `compileSdkVersion = 31`, Device API Level 33 -> Permission denied, no prompt shown either as the target version was lower than Android 13 and the behavior is same as [described by Android](https://developer.android.com/develop/ui/views/notifications/notification-permission#new-apps).
- `targetSdkVersion = 33`, `compileSdkVersion = 33`, Device API Level 30 -> Permission granted by default
- `targetSdkVersion = 33`, `compileSdkVersion = 33`, Device API Level 33 -> Permission prompt shown when requested

Based on these results, Android seems to allow adding `POST_NOTIFICATIONS` for older compile versions without much issues. So I'll vote to keep this change and add details in docs to remove permission using `tools:node` if needed by any customer [like FCM](https://firebase.google.com/docs/cloud-messaging/android/client#optional_remove_post_notifications_permission).

Reply:
Sounds good, thank you for researching more into it.